src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -499,8 +497,12 @@ protected override void GetDependenciesDueToMethodCodePresenceInternal(ref Depen
             {
                 if (FlowAnnotations.RequiresDataflowAnalysis(method))
                 {
-                    dependencies = dependencies ?? new DependencyList();
-                    dependencies.Add(factory.DataflowAnalyzedMethod(methodIL.GetMethodILDefinition()), "Method has annotated parameters");
+                    MethodIL methodILDefinition = methodIL.GetMethodILDefinition();
+                    if (!CompilerGeneratedState.IsNestedFunctionOrStateMachineMember(methodILDefinition.OwningMethod))
+                    {
```

Comment:
@MichalStrehovsky I would welcome opinions on how to do this better. Basically we don't want to run data flow on compiler generated methods (as they are now processed as part of the parent method). Currently this change avoids adding the DataflowAnalyzedMethod node into the graph in this case, but it means that there are 3 places where we have to check for this.

Alternatives:
* Modify the DataflowAnalyzedMethod to "do nothing" in this case - cons, it would like as if we did data flow by just looking at the graph.
* Modify the factory.DataflowAnalyzedMethod - but this is tricky as it can't really return `null`
* Hide this inside `ReflectionMethodBodyScanner` - meaning the node will be added into the graph and it will trigger data flow, but the data flow will be a NOP.

Reply:
Add a `private void AddDataflowDependencyIfNeeded(ref DependencyList dependencies, NodeFactory factory, MethodIL methodIL, string reason)` method to `UsageBasedMetadataManager` and call it from the 3 places?